### PR TITLE
add abort on shutdown to Baker/FBXBaker/TextureBaker

### PR DESCRIPTION
**Pull request by birarda**

### CODE REVIEW

File: assignment-client/src/assets/AssetServer.cpp [hunk 1]
```diff
@@ -310,6 +273,28 @@ AssetServer::AssetServer(ReceivedMessage& message) :
...
+    // remove pending transfer tasks
+    _transferTaskPool.clear();
+
+    // abort each of our still running bake tasks, remove pending bakes that were never put on the thread pool
+    auto it = _pendingBakes.begin();
+    while (it != _pendingBakes.end()) {
+        auto pendingRunnable =  _bakingTaskPool.tryTake(it.value().get());
+
+        if (pendingRunnable) {
+            _pendingBakes.erase(it);
```
Comment (Atlante45) [inline assignment-client/src/assets/AssetServer.cpp]: `it = _pendingBakes.erase(it);`

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3445/HighFidelity-Beta-PR11404-e1d79ee8f4a8ed0e1f605d57771897366314d94c.dmg

File: assignment-client/src/assets/AssetServer.cpp [hunk 2]
```diff
@@ -310,6 +273,28 @@ AssetServer::AssetServer(ReceivedMessage& message) :
...
 
 void AssetServer::aboutToFinish() {
+
+    // remove pending transfer tasks
+    _transferTaskPool.clear();
+
+    // abort each of our still running bake tasks, remove pending bakes that were never put on the thread pool
+    auto it = _pendingBakes.begin();
+    while (it != _pendingBakes.end()) {
+        auto pendingRunnable =  _bakingTaskPool.tryTake(it.value().get());
```
Comment (Atlante45) [inline assignment-client/src/assets/AssetServer.cpp]: I'd usually use `it->get()` here, I find it more readable, but I'm ok with that too.

File: libraries/image/src/image/Image.cpp [hunk 3]
```diff
@@ -439,14 +452,20 @@ void generateMips(gpu::Texture* texture, QImage& image, int face = -1) {
...
     class SequentialTaskDispatcher : public nvtt::TaskDispatcher {
     public:
+        SequentialTaskDispatcher(const std::atomic<bool>& abortProcessing) : _abortProcessing(abortProcessing) {};
+
+        const std::atomic<bool>& _abortProcessing;
+
         virtual void dispatch(nvtt::Task* task, void* context, int count) override {
             for (int i = 0; i < count; i++) {
-                task(context, i);
+                if (!_abortProcessing.load()) {
```
Comment (Atlante45) [inline libraries/image/src/image/Image.cpp]: You should break when that's true instead.
Otherwise, you run through the loop checking abort every time.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3449/HighFidelity-Beta-PR11404-a8ac015bc4e350ffc768699633f757af51a1f03c.dmg

**Pull request merged**